### PR TITLE
feat(drone): Drone provisioner

### LINKED ISSUE #62: feat(Done): Create a provisioner for the Drone System (server and runners)

### PR DESCRIPTION
**Pull request by nothingmn**
Fixes #62

### CODE REVIEW

File: packages/drone/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Change Log
```
Comment (nsainaney) [inline packages/drone/CHANGELOG.md]: Did this file change? Need not be here as its auto generated

File: packages/drone/c6o/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+apiVersion: system.codezero.io/v1
```
Comment (nsainaney) [inline packages/drone/c6o/manifest.yaml]: I would have expected an app with an array of editions. See the samples folder in czdev (those are JSON but these should be YAML). So technically, this file should be called manifests (plural)

File: packages/drone/k8s/latest/3-server-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: server
+  template:
+    metadata:
+      labels:
+        app: server
+        app.kubernetes.io/managed-by: codezero
+        system.codezero.io/edition: latest
```
Comment (nsainaney) [inline packages/drone/k8s/latest/3-server-deployment.yaml]: Do not add edition here - that'll be set by provisioner

File: packages/drone/k8s/latest/4-runner-deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: runner
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: runner
+        app.kubernetes.io/managed-by: codezero
+        system.codezero.io/edition: latest
```
Comment (nsainaney) [inline packages/drone/k8s/latest/4-runner-deployment.yaml]: Do not set editions here

File: packages/drone/k8s/latest/5-service.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: {{namespace}}
+  name: server
+  labels:
+    name: drone-server
+    role: server
+    system.codezero.io/app: drone-server
+    system.codezero.io/edition: latest
```
Comment (nsainaney) [inline packages/drone/k8s/latest/5-service.yaml]: Remove editions

File: packages/drone/package.json [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  "homepage": "https://github.com/c6o/provisioners#readme",
+  "dependencies": {
+    "@c6o/kubeclient": "^0.0.6",
+    "@provisioner/common": "^0.0.7",
+    "lit-element": "^2.3.1",
+    "mixwith": "^0.1.1",
+    "parcel-bundler": "^1.12.4",
+    "tslib": "^1.11.1"
+  },
+  "gitHead": "db2f853696fe6d336812a46d1cac44855aec9dfa"
```
Comment (nsainaney) [inline packages/drone/package.json]: I've learnt that gitHead need not be checked in

File: packages/drone/src/mixins/createInquire.ts [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+                validate: (githubClientId) => (githubClientId !== '' ? true : ''),
+                askAnswered: true,
+                when: input => input.scmChoice === 'GitHub'
+            },
+            {
+                type: 'input',
+                name: 'githubClientSecret',
+                message: 'GitHub Client Secret:',
+                validate: (githubClientSecret) => (githubClientSecret !== '' ? true : ''),
+                askAnswered: true,
```
Comment (nsainaney) [inline packages/drone/src/mixins/createInquire.ts]: Should we be asking if this is set in the manifest? Good for debugging

File: packages/drone/src/ui/Main.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        // set defaults
+        const edition = this.mediator.applicationSpec.metadata.labels['system.codezero.io/edition']
+        this.serviceSpec.edition = edition
+        this.serviceSpec.storageSize = this.serviceSpec.storageSize || '5Gi'
+        this.serviceSpec.scmChoice = this.serviceSpec.scmChoice || this.scmChoices[0]
+    }
+
+    async end() {
+
+        //our scm provider names are templated based on the dropdown list, just replace the spaces in the name with -
```
Comment (nsainaney) [inline packages/drone/src/ui/Main.ts]: Shouldn't this be checkboxes? Can't user add more than one?

Comment (nsainaney) [reply packages/drone/src/ui/Main.ts]: You can pass an array to the mediator if more than one checked

File: packages/drone/src/ui/index.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+export * from './Main'
```
Comment (nsainaney) [inline packages/drone/src/ui/index.ts]: Arrrg...TitleCase! You win this one Rob....

**Review by nsainaney — CHANGES_REQUESTED** [review]

File: packages/drone/CHANGELOG.md (hunk 1 above)
Comment (nothingmn) [reply packages/drone/CHANGELOG.md]: I removed this file.  Was auto-generated and not necessary.

File: packages/drone/c6o/manifest.yaml (hunk 2 above)
Comment (nothingmn) [reply packages/drone/c6o/manifest.yaml]: I removed this  file for now.  It needs to be part of a larger conversation.

File: packages/drone/src/ui/Main.ts (hunk 8 above)
Comment (nothingmn) [reply packages/drone/src/ui/Main.ts]: From what I can tell, you can only set it up against a single SCM.

File: packages/drone/src/ui/index.ts (hunk 9 above)
Comment (nothingmn) [reply packages/drone/src/ui/index.ts]: Renamed!

File: packages/drone/src/mixins/createInquire.ts (hunk 7 above)
Comment (nothingmn) [reply packages/drone/src/mixins/createInquire.ts]: Agreed.  Updated

**Review by nsainaney — APPROVED** [review]

**Pull request merged**